test/integration/rebalancing_test.lua

```diff
@@ -47,91 +64,141 @@ end
-                edit_replicaset(
-                    uuid: $uuid
-                    roles: []
-                )
-            }
-        ]],
+        query = [[mutation ($uuid: String! $roles: [String!]) {
+            edit_replicaset(uuid: $uuid roles: $roles)
+        }]],
         variables = {
             uuid = srv.replicaset_uuid,
+            roles = roles,
         },
     })
 end
 
-local function expel_server(srv)
-    g.cluster.main_server:graphql({
-        query = [[
-            mutation ($uuid: String!) {
-                expel_server(uuid: $uuid)
-            }
-        ]],
-        variables = {
-            uuid = srv.instance_uuid,
-        },
-    })
+local function expel_sA1()
+    return g.sB1:graphql({query = string.format(
+        'mutation{ expel_server(uuid: %q) }',
+        g.sA1.instance_uuid
+    )})
+end
+
+local function get(srv, i)
+    return srv.net_box:eval(
+        'return require("vshard").router.callro(...)',
+        {i, 'box.space.test:get', {i}}
+    )
 end
 
 function g.test()
-    local srv = g.cluster:server('storage-1')
+    t.assert_equals(g.sA1.net_box:call('box.space.test:len'), 1500)
+    t.assert_equals(g.sB1.net_box:call('box.space.test:len'), 1500)
+
+    -- Rebalancer runs on sA1
+    g.sA1.net_box:eval([[assert(vshard.storage.internal.rebalancer_fiber ~= nil)]])
+    g.sB1.net_box:eval([[assert(vshard.storage.internal.rebalancer_fiber == nil)]])
 
     -- Can't disable vshard-storage role with non-zero weight
     t.assert_error_msg_contains(
-        "replicasets[bbbbbbbb-0000-0000-0000-000000000000]" ..
+        "replicasets[aaaaaaaa-0000-0000-0000-000000000000]" ..
         " is a vshard-storage which can't be removed",
-        disable_storage_role, srv
+        set_roles, g.sA1, {}
     )
 
     -- It's prohibited to expel storage with non-zero weight
     t.assert_error_msg_contains(
-        "replicasets[bbbbbbbb-0000-0000-0000-000000000000]" ..
+        "replicasets[aaaaaaaa-0000-0000-0000-000000000000]" ..
         " is a vshard-storage which can't be removed",
-        expel_server, srv
+        expel_sA1
     )
 
-    set_weight(srv, 0)
+    g.sA1.net_box:call('vshard.storage.rebalancer_disable')
+    set_weight(g.sA1, 0)
 
     -- Can't disable vshard-storage role until rebalancing finishes
     t.assert_error_msg_contains(
-        "replicasets[bbbbbbbb-0000-0000-0000-000000000000]" ..
+        "replicasets[aaaaaaaa-0000-0000-0000-000000000000]" ..
         " rebalancing isn't finished yet",
-        disable_storage_role, srv
+        set_roles, g.sA1, {}
     )
 
     -- It's prohibited to expel storage until rebalancing finishes
-    t.assert_error_msg_contains(
-        "replicasets[bbbbbbbb-0000-0000-0000-000000000000]" ..
+    t.assert_error_msg_equals(
+        "replicasets[aaaaaaaa-0000-0000-0000-000000000000]" ..
         " rebalancing isn't finished yet",
-        expel_server, srv
+        expel_sA1
     )
 
     -- Speed up rebalancing
```

Comment:
It's better to call it "force", not "speedup"

Reply:
Acoording to the rule "comment why, not how" I've removed it completely